### PR TITLE
Enhance makefile for code coverage and add testall for all testing

### PR DESCRIPTION
**Pull request by chenziliang**
Fix one warning reported by "go vet" in firehoseclient.go which is
unreachable line of code
	modified:   Makefile
	modified:   firehoseclient/firehoseclient.go

### CODE REVIEW

**Comment by chenziliang** [comment]
Developer can run "make testall" to do all testing now: unit test, race condition test, vet and code coverage". For now, our code coverage looks fantastic.

File: firehoseclient/firehoseclient.go [hunk 1]
```diff
@@ -44,7 +44,6 @@ func (f *FirehoseNozzle) routeEvent() error {
 			return err
 		}
 	}
-	return nil
```
Comment (chenziliang) [inline firehoseclient/firehoseclient.go]: Remove this unreachable line reported by "go vet"

File: Makefile [hunk 2]
```diff
@@ -35,31 +38,41 @@ build-linux:
...
+	go build -o splunk-firehose-nozzle  -ldflags ${LDFLAGS} ./main.go
 
 PKGS=$(shell go list ./... | grep -v vendor | grep -v scripts | grep -v testing | grep -v "splunk-firehose-nozzle$$")
 
+
+testall: test vet race cov
+
 test:
-	go test ${PKGS}
+	@go test ${PKGS}
```
Comment (chenziliang) [inline Makefile]: Suppress the verbose output for easier life of developer

File: Makefile [hunk 3]
```diff
@@ -35,31 +38,41 @@ build-linux:
...
 
 vet:
-	go vet ${PKGS}
+	@go vet ${PKGS}
 
 race:
-	go test -race ${PKGS}
+	@go test -race ${PKGS}
 
+cov:
```
Comment (chenziliang) [inline Makefile]: Add coverage testing

Comment (luckyj5) [inline Makefile]: Looks great.

**Pull request merged**